include/arch/arm/aarch32/irq.h

```diff
@@ -109,14 +109,12 @@ extern void z_arm_interrupt_init(void);
 ({ \
```

Comment:
Since you're getting rid of the return value, there's no need to use the GCC extension for compound expressions https://gcc.gnu.org/onlinedocs/gcc/Statement-Exprs.html and can simplify this macro.

Reply:
> Unfortunately, the API definition in `include/irq.h` specify a return value.
> 
> This is some historical cruft AFAIK and everyone ignores it.
> 
> Still, I would like to see the main API adjusted before making this change.
> 
> What warnings are you currently seeing?

With some highly nested macro's I see:

```
/home/galak/git/zephyr/drivers/gpio/gpio_mcux_igpio.c: In function 'mcux_igpio_0_init':
/home/galak/git/zephyr/samples/hello_world/build/zephyr/include/generated/devicetree_unfixed.h:593:54: warning: statement with no effect [-Wunused-value]
  593 | #define DT_N_S_soc_S_gpio_401b8000_IRQ_IDX_1_VAL_irq 71
      |                                                      ^~
/home/galak/git/zephyr/include/sys/util.h:369:26: note: in definition of macro '__DEBRACKET'
  369 | #define __DEBRACKET(...) __VA_ARGS__
      |                          ^~~~~~~~~~~
/home/galak/git/zephyr/include/sys/util.h:366:2: note: in expansion of macro '__GET_ARG2_DEBRACKET'
  366 |  __GET_ARG2_DEBRACKET(one_or_two_args _if_code, _else_code)
      |  ^~~~~~~~~~~~~~~~~~~~
/home/galak/git/zephyr/include/sys/util.h:324:2: note: in expansion of macro '__COND_CODE'
  324 |  __COND_CODE(_XXXX##_flag, _if_1_code, _else_code)
      |  ^~~~~~~~~~~
/home/galak/git/zephyr/include/sys/util.h:321:2: note: in expansion of macro 'Z_COND_CODE_1'
  321 |  Z_COND_CODE_1(_flag, _if_1_code, _else_code)
      |  ^~~~~~~~~~~~~
/home/galak/git/zephyr/include/devicetree.h:1403:2: note: in expansion of macro 'COND_CODE_1'
 1403 |  COND_CODE_1(DT_HAS_COMPAT(DT_DRV_COMPAT), \
      |  ^~~~~~~~~~~
/home/galak/git/zephyr/include/sys/util.h:374:53: note: in expansion of macro '__DEBRACKET'
  374 | #define __GET_ARG2_DEBRACKET(ignore_this, val, ...) __DEBRACKET val
      |                                                     ^~~~~~~~~~~
/home/galak/git/zephyr/include/sys/util.h:366:2: note: in expansion of macro '__GET_ARG2_DEBRACKET'
  366 |  __GET_ARG2_DEBRACKET(one_or_two_args _if_code, _else_code)
      |  ^~~~~~~~~~~~~~~~~~~~
/home/galak/git/zephyr/include/sys/util.h:324:2: note: in expansion of macro '__COND_CODE'
  324 |  __COND_CODE(_XXXX##_flag, _if_1_code, _else_code)
      |  ^~~~~~~~~~~
/home/galak/git/zephyr/include/sys/util.h:321:2: note: in expansion of macro 'Z_COND_CODE_1'
  321 |  Z_COND_CODE_1(_flag, _if_1_code, _else_code)
      |  ^~~~~~~~~~~~~
/home/galak/git/zephyr/include/sys/util.h:344:2: note: in expansion of macro 'COND_CODE_1'
  344 |  COND_CODE_1(_flag, _code, ())
      |  ^~~~~~~~~~~
/home/galak/git/zephyr/drivers/gpio/gpio_mcux_igpio.c:258:3: note: in expansion of macro 'IF_ENABLED'
  258 |   IF_ENABLED(DT_INST_IRQ_HAS_IDX(n, 1),   \
      |   ^~~~~~~~~~
/home/galak/git/zephyr/include/irq.h:51:2: note: in expansion of macro 'ARCH_IRQ_CONNECT'
   51 |  ARCH_IRQ_CONNECT(irq_p, priority_p, isr_p, isr_param_p, flags_p)
      |  ^~~~~~~~~~~~~~~~
/home/galak/git/zephyr/drivers/gpio/gpio_mcux_igpio.c:228:2: note: in expansion of macro 'IRQ_CONNECT'
  228 |  IRQ_CONNECT(DT_INST_IRQ_BY_IDX(n, i, irq),   \
      |  ^~~~~~~~~~~
/home/galak/git/zephyr/include/devicetree.h:1494:38: note: in expansion of macro 'DT_N_S_soc_S_gpio_401b8000_IRQ_IDX_1_VAL_irq'
 1494 | #define DT_CAT(node_id, prop_suffix) node_id##prop_suffix
      |                                      ^~~~~~~
/home/galak/git/zephyr/include/devicetree.h:781:2: note: in expansion of macro 'DT_CAT'
  781 |  DT_CAT(node_id, _IRQ_IDX_##idx##_VAL_##cell)
      |  ^~~~~~
/home/galak/git/zephyr/include/devicetree.h:1323:2: note: in expansion of macro 'DT_IRQ_BY_IDX'
 1323 |  DT_IRQ_BY_IDX(DT_DRV_INST(inst), idx, cell)
      |  ^~~~~~~~~~~~~
/home/galak/git/zephyr/include/sys/util.h:413:36: note: in expansion of macro 'DT_N_INST_0_nxp_imx_gpio'
  413 | #define UTIL_PRIMITIVE_CAT(a, ...) a##__VA_ARGS__
      |                                    ^
/home/galak/git/zephyr/include/sys/util.h:412:26: note: in expansion of macro 'UTIL_PRIMITIVE_CAT'
  412 | #define UTIL_CAT(a, ...) UTIL_PRIMITIVE_CAT(a, __VA_ARGS__)
      |                          ^~~~~~~~~~~~~~~~~~
/home/galak/git/zephyr/include/devicetree.h:213:31: note: in expansion of macro 'UTIL_CAT'
  213 | #define DT_INST(inst, compat) UTIL_CAT(DT_N_INST, DT_DASH(inst, compat))
      |                               ^~~~~~~~
/home/galak/git/zephyr/include/devicetree.h:1116:27: note: in expansion of macro 'DT_INST'
 1116 | #define DT_DRV_INST(inst) DT_INST(inst, DT_DRV_COMPAT)
      |                           ^~~~~~~
/home/galak/git/zephyr/include/devicetree.h:1323:16: note: in expansion of macro 'DT_DRV_INST'
 1323 |  DT_IRQ_BY_IDX(DT_DRV_INST(inst), idx, cell)
      |                ^~~~~~~~~~~
/home/galak/git/zephyr/drivers/gpio/gpio_mcux_igpio.c:228:14: note: in expansion of macro 'DT_INST_IRQ_BY_IDX'
  228 |  IRQ_CONNECT(DT_INST_IRQ_BY_IDX(n, i, irq),   \
      |              ^~~~~~~~~~~~~~~~~~
/home/galak/git/zephyr/drivers/gpio/gpio_mcux_igpio.c:259:7: note: in expansion of macro 'MCUX_IGPIO_IRQ_INIT'
  259 |       MCUX_IGPIO_IRQ_INIT(n, 1);)   \
      |       ^~~~~~~~~~~~~~~~~~~
/home/galak/git/zephyr/samples/hello_world/build/zephyr/include/generated/devicetree_unfixed.h:3096:31: note: in expansion of macro 'MCUX_IGPIO_INIT'
 3096 | #define DT_FOREACH_IMPL_3(fn) fn(0); fn(1); fn(2);
      |                               ^~
/home/galak/git/zephyr/samples/hello_world/build/zephyr/include/generated/devicetree_unfixed.h:3152:42: note: in expansion of macro 'DT_FOREACH_IMPL_3'
 3152 | #define DT_FOREACH_INST_nxp_imx_gpio(fn) DT_FOREACH_IMPL_3(fn)
      |                                          ^~~~~~~~~~~~~~~~~
/home/galak/git/zephyr/include/sys/util.h:413:36: note: in expansion of macro 'DT_FOREACH_INST_nxp_imx_gpio'
  413 | #define UTIL_PRIMITIVE_CAT(a, ...) a##__VA_ARGS__
      |                                    ^
/home/galak/git/zephyr/drivers/gpio/gpio_mcux_igpio.c:264:1: note: in expansion of macro 'DT_INST_FOREACH'
  264 | DT_INST_FOREACH(MCUX_IGPIO_INIT)
      | ^~~~~~~~~~~~~~~
```

Reply:
Ah, I don't think I realized the `( )` in that macro. That happens to be interplaying with my use of IF_ENABLED().